module/zfs/dsl_pool.c

```diff
@@ -965,18 +965,12 @@ dsl_pool_need_dirty_delay(dsl_pool_t *dp)
 	uint64_t delay_min_bytes =
 	    zfs_dirty_data_max * zfs_delay_min_dirty_percent / 100;
 
-	mutex_enter(&dp->dp_lock);
-	uint64_t dirty = dp->dp_dirty_total;
-	mutex_exit(&dp->dp_lock);
-
-	return (dirty > delay_min_bytes);
+	return (dp->dp_dirty_total > delay_min_bytes);
```

Comment:
No objection, but can you just add a small comment here to make it clear that the `&dp->dp_lock` is intentionally not taken here as an optimization and why it's safe to do so.

Reply:
@behlendorf Sure. Added a commend.